portfolio/src/main/webapp/task.js

```diff
@@ -6,20 +6,130 @@
+  constructor(name, description, duration, taskPriority) {
     this.name = name;
     this.description = description;
     this.duration = duration;
-    this.priority = priority;
+    this.taskPriority = taskPriority;
   }
 }
+
+/**
+ * This method grabs the required inputs from the front end of the page that
+ * will populate the Task's information. It requires that the front end fields
+ * for name, and duration to be passed in. Once it checks for the required
+ * inputs, a JavaScript task class is created and passed into the
+ * updateTaskList function.
+ */
+function createNewTask() {
+  const name = document.getElementById('new-task-name').value;
+  const description = document.getElementById('new-task-description').value;
+  const length = document.getElementById('new-task-estimated-length').value;
+  const lengthUnit =
+      document.getElementById('new-task-estimated-length-unit').value;
+  const priority = document.getElementById('new-task-priority').value;
+
+  if (name == '') {
+    return;
+  }
+
+  if (parseInt(length) <= 0 || length == '') {
+    return;
+  }
+
+  const newTask = new Task(
+      name, description, getDurationMinutes(length, lengthUnit),
+      parseInt(priority));
+  updateTaskList(newTask);
+}
+
+/** Display Task information from user input. */
+function updateTaskList(newTask, lengthUnit) {
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card');
+
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newTask.name;
+  cardBody.appendChild(cardTitle);
+
+  const descriptionText = document.createElement('p');
+  descriptionText.classList.add('card-text');
+  descriptionText.innerText = newTask.description;
+  cardBody.appendChild(descriptionText);
+
+  const duration = document.createElement('p');
+  duration.classList.add('card-text');
+  duration.innerText = newTask.duration;
+  cardBody.appendChild(duration);
+
+  const priority = document.createElement('p');
+  priority.classList.add('card-text');
+  priority.innerText = newTask.taskPriority;
+  cardBody.appendChild(priority);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event';
+  cardBody.appendChild(deleteButton);
+
+  const eventList = document.getElementById('new-task-list');
+  eventList.innterHTML = '';
+  eventList.appendChild(newEventCard);
+
+  // The delete button removes the event's card from the UI.
+  deleteButton.onclick = function(newEventCard) {
+    newEventCard.target.closest('div.card').remove();
+  }
+}
+
+/** Returns the number of minutes from the user's input and unit selection. */
+function getDurationMinutes(duration, unit) {
```

Comment:
Can the method comment explain the type for duration and unit? And what are the acceptable values?

I see that you're taking in string for unit. You might want to consider declaring some constant object so you don't have to hard-code the strings.

ex:

const TIME_UNIT = {
  MINUTES: 'minutes',
  HOURS: 'hours',
};

Reply:
Done.